Backpack/src/main/java/net/skyscanner/backpack/barchart/internal/ChartGraphView.kt

```diff
@@ -0,0 +1,76 @@
+package net.skyscanner.backpack.barchart.internal
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.Gravity
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.barchart.BpkBarChart
+import net.skyscanner.backpack.text.BpkText
+import net.skyscanner.backpack.util.Consumer
+
+@SuppressLint("ViewConstructor")
+internal class ChartGraphView constructor(
+  context: Context,
+  colors: BpkBarChart.Colors,
+  onClick: Consumer<BpkBarChart.Column>
+) : FrameLayout(context), Consumer<List<BpkBarChart.Group>> {
+
+  private val onClickWrapper = { holder: ChartBarHolder ->
+    onClick(holder.model!!)
```

Comment:
Not really. If the item view is not bound there will be no data. By using this operator we declare the contract that only the bound items should be displayed and be clickable.

Reply:
The problem is not the contract, but the exception that will be thrown in case an error happens here. Nullpointer exceptions are very uninformative so if this is the required contract can we throw a better exception?
E.g. `holder.model ?: throw IlegalStateException("View model not bound")`